OOPs2/Report.java

```diff
@@ -0,0 +1,73 @@
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+
+/**
+ * This class Handles Report printing
+ * 
+ * @author Udit Saxena
+ * 
+ */
+public class Report extends FileHandling {
+
+	/**
+	 * Print the Report A using report B
+	 * 
+	 * @param reportB
+	 *            - details of report B
+	 * @param noOfUsers
+	 *            - total number of user participate in Survey
+	 * @param filePathOfA
+	 *            - location of Report A
+	 * @param filePathOfB
+	 *            - location of Report B
+	 */
+	protected void printReportA(String[][] reportB, int noOfUsers,
+			String filePathOfA, String filePathOfB) {
+		clearRecord(filePathOfA);
+		int percentage[] = new int[6];
+		for (int index = 0; index < noOfUsers; index++) {
+			percentage[Integer.parseInt(reportB[index][3])] += 1;
+		}
+		int percent;
+		// will find the percentage of the result
+		for (int index = 1; index < 6; index++) {
+			if (percentage[index] != 0) {
+				percent = (int) ((percentage[index] / (float) noOfUsers) * 100);
+				write(index + " - " + percent + "%", filePathOfA);
```

Comment:
Instead of writing to file for each line, We can create a single string and at the end write it in file.